File: js/groupBy.js
function groupBy(L){
  const res=[]
  let i=0
  while (i<L.length){
    const n=L[i], tmp=[n,[n]]
    while (L[i+1]===n){
      i++
      tmp[1].push(n)
    }
    res.push(tmp)
    i++
  }
 return res
}
function groupBy(L){
  const res=[]
  let i=0
  while (i<L.length){
    const n=L[i], tmp=[n]
    while (L[i+1]===n){
      i++
      tmp.push(n)
    }
    res.push([n,tmp.join('')])
    i++
  }
 return res
}

function groupBy(L){
  const res=[]
  let i=0
  while (i<L.length){
    const n=L[i]
    let c=1
    while (L[i+1]===n){
      i++
      c++
    }
    res.push([n,c])
    i++
  }
 return res
}

File: go/groupBy.go
import "strconv"

func GroupBy(s string) [][]int {
  o := [][]int{}
  i := 0
  l := len(s)
  for i < l {
    n := s[i]
    c := 1
    for i+1 < l && s[i+1] == n {
      i++
      c++
    }
    x,_ := strconv.ParseInt(string(n), 10, 64)
    o = append(o, []int{int(x), c})
    i++
  }
  return o
}
